apps/gateway/src/common/messageSchema.ts

```diff
@@ -51,7 +60,14 @@ export const messageSchema = z
   .or(
     z.object({
       role: z.literal(MessageRole.tool),
-      content: contentSchema,
-      id: z.string(),
+      content: z.array(
+        z.object({
+          type: z.literal(ContentType.toolResult),
+          toolCallId: z.string(),
+          toolName: z.string(),
+          result: z.string(),
+          isError: z.boolean().optional(),
+        }),
+      ),
```

Comment:
A `tool` message should only contain an `id` and `content`, at least according to the openai schema. Why do we need the rest?

Reply:
not true according to ai sdk documentation (and it wasn't working either)